ffish.js Board.moveStack() method returns chess960 style UCI castling move in non-chess960 mode

## Steps to reproduce:
In fairyground, open developer's tool and switch to console, type following command:
```javascript
let aaa=new window.ffishlib.Board("seirawan","4kbnr/pppppppp/8/8/8/8/PPPPPPPP/RNBQKBNR[EH] w KQBCDFGk - 0 1",false);
aaa.pushMoves("e2e4 g8f6 e4e5 f6d5 c2c4 d5b4 a2a3 b4c6 d2d4 e7e6 b1c3h f8e7 g1f3e e7d8 b2b4 c6e7 b1e4 a7a6 h2h4 h7h6 e4b7 e8g8 g2g4 a6a5 b4a5 c7c5 d4c5 g8h7 d1d7 h7h8 a5a6 e7g6 a6a7 d8a5 b7a5 h8h7 g4g5 h6h5 g1g3 g6f4 c1f4 g7g6 a1b1 h7g8 b1b8 g8g7 b8f8 g7f8 a7a8q f8g7 a8e8");
console.log(aaa.moveStack());
aaa.push("g7h7");
console.log(aaa.moveStack());
aaa.pop();
console.log(aaa.moveStack());
```
The result is:

<img width="607" height="464" alt="Image" src="https://github.com/user-attachments/assets/a1dd7549-5de3-4f5e-ad7f-15a2748fc62b" />

Pay attention to the move within the red box. Before move `g7h7` is pushed, it is `e8g8`, which is the correct UCI move for kingside castling (BLACK king moves from `e8` to `g8`). However, after move `g7h7` is pushed, the move at this position becomes `e8h8`, which is a chess960 style UCI move of kingside castling. This is problematic as the board is created without chess960, and thus such move is illegal, like this:

<img width="638" height="364" alt="Image" src="https://github.com/user-attachments/assets/983a69c4-915d-4620-98b8-4dbac3a7546c" />

And after undoing move `g7h7` by calling `Board.pop()`, it becomes correct. Therefore something goes wrong in `Board.moveStack()`. Based on the code in:

https://github.com/fairy-stockfish/Fairy-Stockfish/blob/f4f8ef1a03b308b5831244c19c54e26eca3c8bbf/src/ffishjs.cpp#L355C1-L364C1

Either the `Position pos` structure is messed up during the process above, or `UCI::move` function is having problem. I'm not familiar with this part of code, so I raise this issue.

I have the feeling that this can be the same issue caused a server crashes on pychess site lately.
One of our user imported this standard chess game https://www.pychess.org/zfyV1kno
Importing PGN game uses ffish.readGamePGN(pgn) on client side.
I find this in the logs:
```
Logplex-Msg-Count: 47 d.77a43bf3-3d54-411f-b1ca-d055f82453d2
1043 <190>1 2025-11-09T11:11:39.722903+00:00 host app web.1 - +0000 2025-11-09 11:11:39.721 [ERROR] root:1098 Exception in create_steps() zfyV1kno chess rnbqkbnr/pppppppp/8/8/8/8/PPPPPPPP/RNBQKBNR w KQkq - 0 1 e8h8 ['e2e4', 'c7c5', 'g1f3', 'd7d6', 'd2d4', 'c5d4', 'f3d4', 'g8f6', 'b1c3', 'a7a6', 'h2h3', 'e7e5', 'd4e2', 'h7h5', 'g2g3', 'f8e7', 'f1g2', 'b7b5', 'c3d5', 'f6d5', 'd1d5', 'a8a7', 'c1e3', 'a7b7', 'd5d2', 'b8d7', 'b2b3', 'd7f6', 'e1c1', 'e8h8', 'g3g4', 'h5h4', 'h1g1', 'f6h7', 'f2f4', 'e5f4', 'e2f4', 'b7d7', 'c1b1', 'e7g5', 'g1f1', 'f8e8', 'd2f2', 'c8b7', 'e3b6', 'd8e7', 'f4d5', 'b7d5', 'd1d5', 'g5f6', 'b6d4', 'f6d4', 'f2d4', 'h7g5', 'b1b2', 'g5e6', 'd4b6', 'e7d8', 'b6d8', 'e8d8', 'e4e5', 'e6c7', 'd5d6', 'd7d6', 'e5d6', 'd8d6', 'b2c3', 'c7e6', 'f1f2', 'g8f8', 'c3b4', 'g7g5', 'b4a5', 'e6f4', 'g2f1', 'f8e7', 'b3b4', 'd6c6', 'a2a4', 'b5a4', 'c2c4', 'f7f5', 'g4f5', 'c6f6', 'c4c5', 'f6f5', 'f2f3', 'f5f6', 'f1a6', 'f6e6', 'b4b5', 'e6e5', 'c5c6', 'e7d6', 'a6b7', 'd6c7', 'a5a6', 'f4d5', 'b5b6', 'c7d6', 'a6a7', 'e5e7', 'f3d3', 'e7e8']
89 <190>1 2025-11-09T11:11:39.722928+00:00 host app web.1 - ValueError: Invalid move 'e8h8'
58 <190>1 2025-11-09T11:11:39.722928+00:00 host app web.1 -
126 <190>1 2025-11-09T11:11:39.722929+00:00 host app web.1 - The above exception was the direct cause of the following exception:
58 <190>1 2025-11-09T11:11:39.722929+00:00 host app web.1 -
92 <190>1 2025-11-09T11:11:39.722929+00:00 host app web.1 - Traceback (most recent call last):
117 <190>1 2025-11-09T11:11:39.72293+00:00 host app web.1 -   File "/app/server/fairy/fairy_board.py", line 180, in push
88 <190>1 2025-11-09T11:11:39.72293+00:00 host app web.1 -     self.fen = self.sf.get_fen(
88 <190>1 2025-11-09T11:11:39.72293+00:00 host app web.1 -                ~~~~~~~~~~~~~~~^
78 <190>1 2025-11-09T11:11:39.72293+00:00 host app web.1 -         self.variant,
79 <190>1 2025-11-09T11:11:39.722931+00:00 host app web.1 -         ^^^^^^^^^^^^^
77 <190>1 2025-11-09T11:11:39.722931+00:00 host app web.1 -     ...<5 lines>...
85 <190>1 2025-11-09T11:11:39.722931+00:00 host app web.1 -         self.count_started,
85 <190>1 2025-11-09T11:11:39.722931+00:00 host app web.1 -         ^^^^^^^^^^^^^^^^^^^
63 <190>1 2025-11-09T11:11:39.722931+00:00 host app web.1 -     )
63 <190>1 2025-11-09T11:11:39.722932+00:00 host app web.1 -     ^
138 <190>1 2025-11-09T11:11:39.722932+00:00 host app web.1 - SystemError: <built-in function get_fen> returned a result with an exception set
58 <190>1 2025-11-09T11:11:39.722932+00:00 host app web.1 -
125 <190>1 2025-11-09T11:11:39.722933+00:00 host app web.1 - During handling of the above exception, another exception occurred:
58 <190>1 2025-11-09T11:11:39.722933+00:00 host app web.1 -
89 <190>1 2025-11-09T11:11:39.722933+00:00 host app web.1 - ValueError: Invalid move 'e7e8'
58 <190>1 2025-11-09T11:11:39.722933+00:00 host app web.1 -
126 <190>1 2025-11-09T11:11:39.722933+00:00 host app web.1 - The above exception was the direct cause of the following exception:
58 <190>1 2025-11-09T11:11:39.722933+00:00 host app web.1 -
92 <190>1 2025-11-09T11:11:39.722933+00:00 host app web.1 - Traceback (most recent call last):
114 <190>1 2025-11-09T11:11:39.722934+00:00 host app web.1 -   File "/app/server/game.py", line 1068, in create_steps
97 <190>1 2025-11-09T11:11:39.722934+00:00 host app web.1 -     self.board.push(move, append=False)
97 <190>1 2025-11-09T11:11:39.722934+00:00 host app web.1 -     ~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^
118 <190>1 2025-11-09T11:11:39.722934+00:00 host app web.1 -   File "/app/server/fairy/fairy_board.py", line 190, in push
72 <190>1 2025-11-09T11:11:39.722934+00:00 host app web.1 -     self.pop()
72 <190>1 2025-11-09T11:11:39.722934+00:00 host app web.1 -     ~~~~~~~~^^
117 <190>1 2025-11-09T11:11:39.722934+00:00 host app web.1 -   File "/app/server/fairy/fairy_board.py", line 207, in pop
89 <190>1 2025-11-09T11:11:39.722935+00:00 host app web.1 -     self.fen = self.sf.get_fen(
89 <190>1 2025-11-09T11:11:39.722935+00:00 host app web.1 -                ~~~~~~~~~~~~~~~^
79 <190>1 2025-11-09T11:11:39.722935+00:00 host app web.1 -         self.variant,
79 <190>1 2025-11-09T11:11:39.722935+00:00 host app web.1 -         ^^^^^^^^^^^^^
77 <190>1 2025-11-09T11:11:39.722935+00:00 host app web.1 -     ...<5 lines>...
85 <190>1 2025-11-09T11:11:39.722935+00:00 host app web.1 -         self.count_started,
85 <190>1 2025-11-09T11:11:39.722935+00:00 host app web.1 -         ^^^^^^^^^^^^^^^^^^^
63 <190>1 2025-11-09T11:11:39.722935+00:00 host app web.1 -     )
63 <190>1 2025-11-09T11:11:39.722936+00:00 host app web.1 -     ^
138 <190>1 2025-11-09T11:11:39.722936+00:00 host app web.1 - SystemError: <built-in function get_fen> returned a result with an exception set
137 <190>1 2025-11-09T11:11:39.725258+00:00 host app web.1 - +0000 2025-11-09 11:11:39.725 [ERROR] root:827 Exception in game zfyV1kno pgn()
```

--- a/src/ffishjs.cpp
+++ b/src/ffishjs.cpp
@@ -84,6 +84,9 @@ private:
   Position pos;
   Thread* thread;
   std::vector<Move> moveStack;
+  // UCI notation of the moves in moveStack, recorded at the time the move is made,
+  // since conversion of a move to UCI notation can depend on the position, e.g., for castling.
+  std::vector<std::string> uciMoveStack;
   bool is960;
 
 public:
@@ -160,6 +163,7 @@ public:
   void pop() {
     pos.undo_move(this->moveStack.back());
     moveStack.pop_back();
+    uciMoveStack.pop_back();
     states->pop_back();
   }
 
@@ -186,6 +190,7 @@ public:
   void set_fen(std::string fen) {
     resetStates();
     moveStack.clear();
+    uciMoveStack.clear();
     pos.set(v, fen, is960, &states->back(), thread);
   }
 
@@ -349,8 +354,8 @@ public:
 
   std::string move_stack() const {
     std::string moves;
-    for(auto it = std::begin(moveStack); it != std::end(moveStack); ++it) {
-      moves += UCI::move(pos, *it);
+    for(auto it = std::begin(uciMoveStack); it != std::end(uciMoveStack); ++it) {
+      moves += *it;
       moves += DELIM;
     }
     save_pop_back(moves);
@@ -431,6 +436,7 @@ private:
   }
 
   void do_move(Move move) {
+    this->uciMoveStack.emplace_back(UCI::move(pos, move));
     states->emplace_back();
     this->pos.do_move(move, states->back());
     this->moveStack.emplace_back(move);

--- a/tests/js/test.js
+++ b/tests/js/test.js
@@ -623,6 +623,23 @@ describe('board.moveStack()', function () {
     chai.expect(board.moveStack()).to.equal("h5f7");
     board.delete();
   });
+  it("it keeps non-960 castling notation when later moves obscure it (issue #911)", () => {
+    // After castling, the rook returns to the king's origin square e8 and the king
+    // vacates g8, which used to flip the rendered castling move to 960 notation e8h8.
+    let board = new ffish.Board("chess");
+    const chessMoves = "e2e4 e7e5 g1f3 g8f6 f1c4 f8c5 d2d3 e8g8 c1g5 f8e8 b1c3 g8h8 a2a3";
+    board.pushMoves(chessMoves);
+    chai.expect(board.moveStack()).to.equal(chessMoves);
+    board.delete();
+    board = new ffish.Board("seirawan", "4kbnr/pppppppp/8/8/8/8/PPPPPPPP/RNBQKBNR[EH] w KQBCDFGk - 0 1");
+    const moves = "e2e4 g8f6 e4e5 f6d5 c2c4 d5b4 a2a3 b4c6 d2d4 e7e6 b1c3h f8e7 g1f3e e7d8 b2b4 c6e7 " +
+                  "b1e4 a7a6 h2h4 h7h6 e4b7 e8g8 g2g4 a6a5 b4a5 c7c5 d4c5 g8h7 d1d7 h7h8 a5a6 e7g6 " +
+                  "a6a7 d8a5 b7a5 h8h7 g4g5 h6h5 g1g3 g6f4 c1f4 g7g6 a1b1 h7g8 b1b8 g8g7 b8f8 g7f8 " +
+                  "a7a8q f8g7 a8e8 g7h7";
+    board.pushMoves(moves);
+    chai.expect(board.moveStack()).to.equal(moves);
+    board.delete();
+  });
 });
 
 describe('board.pushMoves(uciMoves)', function () {

